src/librustc_resolve/late.rs

```diff
@@ -841,91 +825,84 @@ impl<'a, 'b> LateResolutionVisitor<'a, '_> {
-
-                            // Plain insert (no renaming).
-                            let res = Res::Def(
-                                DefKind::TyParam,
-                                self.r.definitions.local_def_id(param.id),
-                            );
-                            function_type_rib.bindings.insert(ident, res);
-                            self.r.record_partial_res(param.id, PartialRes::new(res));
-                        }
-                        GenericParamKind::Const { .. } => {
-                            let ident = param.ident.modern();
-                            debug!("with_generic_param_rib: {}", param.id);
-
-                            if seen_bindings.contains_key(&ident) {
-                                let span = seen_bindings.get(&ident).unwrap();
-                                let err = ResolutionError::NameAlreadyUsedInParameterList(
-                                    ident.name,
-                                    *span,
-                                );
-                                self.r.report_error(param.ident.span, err);
-                            }
-                            seen_bindings.entry(ident).or_insert(param.ident.span);
-
-                            let res = Res::Def(
-                                DefKind::ConstParam,
-                                self.r.definitions.local_def_id(param.id),
-                            );
-                            function_value_rib.bindings.insert(ident, res);
-                            self.r.record_partial_res(param.id, PartialRes::new(res));
-                        }
+                GenericParamKind::Const { .. } => {
+                    let ident = param.ident.modern();
+                    debug!("with_generic_param_rib: {}", param.id);
+
+                    if seen_bindings.contains_key(&ident) {
+                        let span = seen_bindings.get(&ident).unwrap();
+                        let err = ResolutionError::NameAlreadyUsedInParameterList(
+                            ident.name,
+                            *span,
+                        );
+                        self.r.report_error(param.ident.span, err);
                     }
+                    seen_bindings.entry(ident).or_insert(param.ident.span);
+
+                    let res = Res::Def(
+                        DefKind::ConstParam,
+                        self.r.definitions.local_def_id(param.id),
+                    );
+                    function_value_rib.bindings.insert(ident, res);
+                    self.r.record_partial_res(param.id, PartialRes::new(res));
                 }
-                self.ribs[ValueNS].push(function_value_rib);
-                self.ribs[TypeNS].push(function_type_rib);
-            }
-
-            NoGenericParams => {
-                // Nothing to do.
             }
         }
 
+        self.ribs[ValueNS].push(function_value_rib);
```

Comment:
Would be good to maintain the stack discipline here with:
```rust
         self.with_rib(ValueNS, function_value_rib, |this| {
             this.with_rib(TypeNS, function_type_rib, f);
         });
```

Reply:
I haven't changed this because `with_rib` doesn't take a `Rib`, it takes a `RibKind`, and `function_{value,type}_rib` are `Rib`s, not `RibKind`s.

Reply:
Oh right... 🤦‍♂  that's probably why I didn't change that bit of code in the first place.